src/language-markdown/embed.js

```diff
@@ -12,7 +12,7 @@ function embed(path, print, textToDoc, options) {
 
   if (node.type === "code" && node.lang !== null) {
     // only look for the first string so as to support [markdown-preview-enhanced](https://shd101wyy.github.io/markdown-preview-enhanced/#/code-chunk)
-    const lang = node.lang.split(/\s/, 1)[0];
+    const lang = (node.lang.match(/^[A-Za-z0-9_-]+/) || [])[0];
```

Comment:
`lang` should be a string, we could use `.match(/^[A-Za-z0-9_-]*/)` or `.split(/[^A-Za-z0-9_-]/, 1)` to avoid `undefined` in this case.

Reply:
What about?

```js
const lang = (node.lang.match(/^[A-Za-z0-9_-]+/) || [])[0] || "";
```

or

```js
const langMatch = node.lang.match(/^[A-Za-z0-9_-]+/);
const lang = langMatch ? langMatch[0] : "";
```

Nevertheless, it's interesting that none of the tests fail despite that lang is undefined 🤔

Reply:
Isn't the first one much more complex? 😂 The second one looks good to me.